src/compress/zip/zip.cr

```diff
@@ -27,7 +27,17 @@ require "digest/crc32"
 # NOTE: only compression methods 0 (STORED) and 8 (DEFLATED) are
 # supported. Additionally, ZIP64 is not yet supported.
 module Compress::Zip
-  VERSION                                   =     20_u16
+  VERSION = 20_u8
+
+  {% if flag?(:unix) %}
+    FS_ORIGIN = 3_u8 # Unix
+  {% elsif flag?(:win32) %}
+    FS_ORIGIN = 0_u8 # # MS-DOS, OS/2, or NT FAT
+  {% else %}
+    {% raise "No FS_ORIGIN implementation available" %}
+  {% end %}
```

Comment:
I have some doubts about these hard-coded per-platform values. The specification looks like this mostly depends on the file system, not the operating system. Windows NTFS would be `10` for example.

However, I don't know how this really works, so I might just be mistaken and this is fine.

Reply:
The spec says "operating system or file system".  As far as i know it's used as a hint for line ending encoding in text files or in ancient times EBCEDIC.  It should probably use whatever value winzip uses but I don't have access.